firebase-appdistribution/src/main/java/com/google/firebase/appdistribution/AabUpdater.java

```diff
@@ -89,7 +93,18 @@ UpdateTaskImpl updateAab(@NonNull AppDistributionReleaseInternal newRelease) {
 
       cachedUpdateTask = new UpdateTaskImpl();
       aabReleaseInProgress = newRelease;
-      redirectToPlayForAabUpdate(newRelease.getDownloadUrl());
+      hasBeenSentToPlayForCurrentTask = false;
+
+      // On a background thread, fetch the redirect URL and open it in the Play app
+      runAsyncInTask(executor, () -> fetchDownloadRedirectUrl(newRelease.getDownloadUrl()))
+          .onSuccessTask(
+              executor,
```

Comment:
Is there a reason for this chain of tasks we use this executor but we don't do this elsewhere?

Reply:
Good question. We actually do this inconsistently. For example [ApkUpdater](https://github.com/firebase/firebase-android-sdk/blob/master/firebase-appdistribution/src/main/java/com/google/firebase/appdistribution/ApkUpdater.java#L101) uses its own executor, while [TesterSignInManager](https://github.com/firebase/firebase-android-sdk/blob/master/firebase-appdistribution/src/main/java/com/google/firebase/appdistribution/TesterSignInManager.java#L145) does not (so callbacks run on the main thread).

In this case since we're already using it for the fetchDownloadRedirectUrl it seemed to make sense to use it everywhere. This also makes testing easier for the reasons we were discussing earlier. Robolectric unit tests run on the main thread, so awaiting Tasks to finish blocks execution of callbacks that also run on the main thread.

I think we should come up with a more consistent approach though.